pkg/gameservers/migration.go

```diff
@@ -0,0 +1,194 @@
+	mc.baseLogger.Info("Wait for cache sync")
+	if !cache.WaitForCacheSync(stop, mc.nodeSynced, mc.gameServerSynced, mc.podSynced) {
+		return errors.New("failed to wait for caches to sync")
+	}
+
+	mc.workerqueue.Run(1, stop)
+	return nil
+}
+
+func (mc *MigrationController) loggerForGameServerKey(key string) *logrus.Entry {
+	return logfields.AugmentLogEntry(mc.baseLogger, logfields.GameServerKey, key)
+}
+
+// isActiveGameServerWithNode tests to see if a Pod is active and has a Node assigned.
+func isActiveGameServerWithNode(pod *corev1.Pod) bool {
+	return pod.Spec.NodeName != "" && pod.ObjectMeta.DeletionTimestamp.IsZero() && isGameServerPod(pod)
+}
+
+// syncGameServer will check if the Pod for the GameServer
+// has been migrated to a new node (or a node with the same name, but different address)
+// and will either update it, or move it to Unhealthy, depending on its State
+func (mc *MigrationController) syncGameServer(key string) error {
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		// don't return an error, as we don't want this retried
+		runtime.HandleError(mc.loggerForGameServerKey(key), errors.Wrapf(err, "invalid resource key"))
+		return nil
+	}
+
+	gs, err := mc.gameServerLister.GameServers(namespace).Get(name)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			mc.loggerForGameServerKey(key).Debug("GameServer is no longer available for syncing")
+			return nil
+		}
+		return errors.Wrapf(err, "error retrieving GameServer %s from namespace %s", name, namespace)
+	}
+
+	// Either the address has not been set, or we're being deleted already
+	if gs.Status.NodeName == "" || gs.IsBeingDeleted() || gs.Status.State == agonesv1.GameServerStateUnhealthy {
+		return nil
+	}
+
+	pod, err := mc.podLister.Pods(namespace).Get(name)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			mc.loggerForGameServerKey(key).Debug("Pod is no longer available for syncing")
+			return nil
+		}
+		return errors.Wrapf(err, "error retrieving Pod %s from namespace %s", name, namespace)
+	}
+	if !pod.ObjectMeta.DeletionTimestamp.IsZero() {
+		return nil
+	}
+
+	node, err := mc.nodeLister.Get(pod.Spec.NodeName)
+	if err != nil {
+		return errors.Wrapf(err, "error retrieving node %s for Pod %s", pod.Spec.NodeName, pod.ObjectMeta.Name)
+	}
+	address, err := address(node)
+	if err != nil {
+		return err
+	}
+
+	if pod.Spec.NodeName != gs.Status.NodeName || address != gs.Status.Address {
+		gsCopy := gs.DeepCopy()
+
+		var eventMsg string
+		// If the GameServer has yet to become ready, we will reapply the Address and Port
+		// otherwise, we move it to Unhealthy so that a new GameServer will be recreated.
+		if gsCopy.IsBeforeReady() {
+			gsCopy, err = applyGameServerAddressAndPort(gsCopy, node, pod)
+			if err != nil {
+				return err
+			}
+			eventMsg = "Address updated due to Node migration"
+		} else {
+			gsCopy.Status.State = agonesv1.GameServerStateUnhealthy
+			eventMsg = "Node migration occurred"
+		}
+
```

Comment:
This seems like a good place to have a logger.Info line written

Reply:
Adding a Debug log in here. Doesn't feel like there there should be an Info level event?